client/src/useJobPaginator.js

```diff
@@ -0,0 +1,186 @@
+import { useState } from "react";
+import MagicLabeler from "./MagicLabeler";
+import AjaxHelper from "./AjaxHelper";
+import {
+    countMagicLabels,
+    deleteImage,
+    deletePageValidation,
+    getAddedPages,
+    getNoneRemovedPages,
+    getStatus,
+    getJob,
+    getLabel,
+    putJob,
+    validatePublish,
+    validateSave,
+} from "./JobPaginatorState";
+
+const useJobPaginator = (initialCategory, initialJob) => {
+    const [active] = useState(false);
+    const [currentPage, setCurrentPage] = useState(0);
+    const [zoom, setZoom] = useState(false);
+    const [order, setOrder] = useState([]);
+    const [magicLabelCache, setMagicLabelCache] = useState([]);
+    const [category] = useState(initialCategory);
+    const [job] = useState(initialJob);
+
+    const saveMagicLabels = () => {
+        order.forEach((o, orderIndex) => {
+            if (!getLabel(order, orderIndex)) {
+                setLabel(orderIndex, getMagicLabel(orderIndex));
+            }
+        });
+    };
+
+    const setLabel = (imageNumber, text) => {
+        setMagicLabelCache([]); // clear label cache whenever there is a change
+        if (typeof order[imageNumber] === "undefined") {
+            return;
+        }
+        order[imageNumber].label = text !== null && text.length === 0 ? null : text;
+        setOrder(order);
+        dispatchEvent(new Event("Prep.editted"));
+    };
+
+    const getMagicLabel = (imageNumber) => {
+        const label = getLabel(order, imageNumber);
+
+        if (label) {
+            return label;
+        }
+        if (typeof magicLabelCache[imageNumber] === "undefined") {
+            magicLabelCache[imageNumber] = MagicLabeler.getLabel(imageNumber, getMagicLabel);
+            setMagicLabelCache(magicLabelCache);
+        }
+        return magicLabelCache[imageNumber];
+    };
+
+    const setPage = (page) => {
+        if (page >= 0 && page < order.length) {
+            setCurrentPage(page);
+        }
+    };
+
+    const nextPage = () => {
+        setPage(currentPage + 1);
+    };
+
+    const prevPage = () => {
+        setPage(currentPage - 1);
+    };
+
+    const updatePagesByStatus = (status) => {
+        const {
+            file_problems: { deleted, added },
+        } = status;
+        const message = [];
+        if (deleted.length > 0) {
+            message.push(deleted.length + " file(s) have been removed from the job since the last edit.\n");
+            setOrder(getNoneRemovedPages(order, deleted));
+        }
+
+        if (added.length > 0) {
+            message.push(added.length + " file(s) have been added to the job since the last edit.\n");
+            setOrder(getAddedPages(order, added));
+        }
+        return message.length ? alert(message.join()) : undefined;
```

Comment:
Does it really make sense to have a `return` here at all? While it's a tidy one-liner as written, it's also a bit unusual. I wonder if it would be more straightforward to just have a standard `if (message.length) { alert(message.join()); }` here.